db-tests/src/test/java/TestTriggers.java

```diff
@@ -577,4 +639,124 @@ void shouldIncrementSimulationRevisionOnSimulationUpdate() throws SQLException {
+          .executeQuery(
+              """
+                  INSERT INTO plan_dataset (plan_id, offset_from_plan_start)
+                  VALUES (%s, '0')
+                  RETURNING dataset_id;"""
+                  .formatted(planId)
+          );
+      res.next();
+      final var newDatasetId = res.getInt("dataset_id");
+      res.close();
+
+      assertInstanceOf(Integer.class, newDatasetId);
+
+      final var datasetRes = connection.createStatement()
+          .executeQuery(
+              """
+                  SELECT * FROM dataset
+                  WHERE id = %s;"""
+                  .formatted(newDatasetId)
+          );
+
+      datasetRes.next();
+      assertEquals(newDatasetId, datasetRes.getInt("id"));
+      assertEquals(0, datasetRes.getInt("revision"));
+      datasetRes.close();
+    }
+
+    @Test
+    void shouldCalculatePlanDatasetOffsetOnPlanDatasetInsertWithNonNullDatasetId() throws SQLException {
+
+      final var planRes = connection.createStatement()
+          .executeQuery(
+              """
+                  SELECT * from plan
+                  WHERE id = %s;"""
+                  .formatted(planDatasetId.getLeft())
+          );
+      planRes.next();
+      final var planStartTime = planRes.getTimestamp("start_time");
+      planRes.close();
+
+      final var planDatasetSelectRes = connection.createStatement()
+          .executeQuery(
+              """
+                  SELECT * FROM plan_dataset
+                  WHERE plan_id = %s and dataset_id = %s;"""
+                  .formatted(planDatasetId.getLeft(), planDatasetId.getRight())
+          );
+      planDatasetSelectRes.next();
+      final var offsetFromPlanStart = Duration.parse(planDatasetSelectRes.getString("offset_from_plan_start"));
+      planDatasetSelectRes.close();
+      assertEquals(Duration.ofMillis(0), offsetFromPlanStart);
+
+      final var newPlanId = insertPlan(missionModelId, "2020-1-1 01:00:00");
+
+      final var newPlanRes = connection.createStatement()
+          .executeQuery(
+              """
+                  SELECT * from plan
+                  WHERE id = %s;"""
+                  .formatted(newPlanId)
+          );
+      newPlanRes.next();
+      final var newPlanStartTime = newPlanRes.getTimestamp("start_time");
+      newPlanRes.close();
+
+      final var planDatasetInsertRes = connection.createStatement()
+          .executeQuery(
+              """
+                  INSERT INTO plan_dataset (plan_id, dataset_id)
+                  VALUES (%s, %s)
+                  RETURNING *;"""
+                  .formatted(newPlanId, planDatasetId.getRight())
+          );
+      planDatasetInsertRes.next();
+      final var newOffsetFromPlanStart = Duration.parse(planDatasetInsertRes.getString("offset_from_plan_start"));
+      planDatasetInsertRes.close();
+
+      assertEquals(offsetFromPlanStart.minus(Duration.ofMillis(newPlanStartTime.getTime() - planStartTime.getTime())), newOffsetFromPlanStart);
```

Comment:
Would you mind storing the result of the computation in a variable before this so that the assertion is more clear at a glance?

Reply:
Done